aesthetic_ants/penguin/tile_layer.py

```diff
@@ -121,3 +172,35 @@ def collide_tiles(self, other, _):
                 continue
 
             other.collide_tile(self.tiles[y][x])
+
+    def decay_tiles(self, dt: float, enemies: List[Enemy]):
+        self.decay += DECAY_RATE * dt
+        self.decay += dt * ENEMY_DECAY_INFLUENCE * len(enemies)
+
+        grid_width = len(self.tiles[0])
+        grid_height = len(self.tiles)
+
+        for _ in range(round(self.decay)):
+            self.decay -= 1
+            x = random.randrange(grid_width)
+            y = random.randrange(grid_height)
+            tile = self.tiles[y][x]
+            if tile.tile_type in DECAY_MAP:
+                tile_weakness = 0
+                for offset_x, offset_y in NEARBY_TILES:
+                    # Skip out of bounds tiles
+                    if not 0 <= x + offset_x < grid_width:
+                        continue
+                    if not 0 <= y + offset_y < grid_height:
+                        continue
+
+                    if self.tiles[y + offset_y][x + offset_x].tile_type in WEAK_TILES:
+                        tile_weakness += WEAK_TILE_DECAY_INFLUENCE
```

Comment:
This entire for-loop is just basically `tile_weakness = WEAK_TILE_DECAY_INFLUENCE * (how many tiles around this tile are weak)`.

Would probably not hurt to have a `neighbours(self, x, y)` method either.